### PR TITLE
[17508] Linux CI on Github Actions

### PR DESCRIPTION
**Pull request by Mario-DL**
This PR adds a Linux CI workflow on Github Actions. 
I open as draft until eProsima/eProsima-CI#4 can be merged so that the final branch could be set on the colcon installation step.

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/160/)
* Linux aarch64Build Status
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/88/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/152/)

File: .github/workflows/fastcdr-test.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      uses: eProsima/eProsima-CI/ubuntu/install_colcon@feature/uncrustify
+
+    - name: Fetch repo
+      if: ${{ github.event.inputs.fastcdr_branch}}
+      run: |
+        cd src/Fast-CDR && git checkout ${{ inputs.fastcdr_branch }} && cd ../..
+
+    - name: Build
+      run: |
+        colcon build --event-handlers=console_direct+ --cmake-args ${{ inputs.cmake_args }}
```
Comment (jsantiago-eProsima) [inline .github/workflows/fastcdr-test.yml:77]: For testing maybe it's worth it to hardcode the CMAKE_BUILD_TYPE=RelWithDebInfo like Jenkins is currently doing.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: .github/workflows/fastcdr-test.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      uses: actions/checkout@v3
+      with:
+          path: src/Fast-CDR
+
+    - name: Install Colcon and dependencies
+      uses: eProsima/eProsima-CI/ubuntu/install_colcon@main
+
+    - name: Build
+      run: |
+        colcon build --event-handlers=console_direct+ --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo ${{ inputs.cmake_args }}
```
Comment (jsantiago-eProsima) [inline .github/workflows/fastcdr-test.yml:68]: I think we are missing an EPROSIMA_BUILD_TESTS flag.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/166/)
* Linux aarch64Build Status
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/93/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/158/)

File: .github/workflows/fastcdr-test.yml [hunk 3]
```diff
@@ -60,13 +60,32 @@ jobs:
       with:
           path: src/Fast-CDR
 
-    - name: Install Colcon and dependencies
+    - name: Install Colcon and Dependencies
       uses: eProsima/eProsima-CI/ubuntu/install_colcon@main
 
+    # Temporal step as this platform will be discontinued soon
+    - name: Install Gtest Backwards Compatibility
+      if: ${{ matrix.ubuntu-version == 'ubuntu-20.04' }}
```
Comment (jsantiago-eProsima) [inline .github/workflows/fastcdr-test.yml:68]: Right now Ubuntu 22.04 is doing both installations. You could probably add a similar if clause to the other GTest installation.

File: .github/workflows/fastcdr-test.yml [hunk 4]
```diff
@@ -60,13 +60,32 @@ jobs:
...
+            --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo -DEPROSIMA_BUILD_TESTS=ON \
+            ${{ inputs.cmake_args }}
 
     - name: Run Tests
       run: |
-        colcon test --event-handlers=console_direct+ --packages-select fastcdr --ctest-args ${{ inputs.ctest_args }}
+        colcon test \
+            --event-handlers=console_direct+ \
+            --packages-select fastcdr \
+            --ctest-args ${{ inputs.ctest_args }}
```
Comment (jsantiago-eProsima) [inline .github/workflows/fastcdr-test.yml:91]: Have you tried if the workflow fails with failing tests?

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: .github/workflows/fastcdr-test.yml (hunk 4 above)
Comment (Mario-DL) [reply .github/workflows/fastcdr-test.yml]: Nice catch !
I added `--return-code-on-test-failure`.
Checked [here](https://github.com/Mario-DL/Fast-CDR/actions/runs/4445155806/jobs/7804016395)

**Review by jsantiago-eProsima — APPROVED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/168/)
* Linux aarch64Build Status
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/94/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/160/)

**Pull request merged**